buttercup/cogs/rules.py

```diff
@@ -53,6 +85,12 @@ async def _rules(self, ctx: SlashContext, subreddit: str) -> None:
         embed = Embed(title=i18n["rules"]["embed_title"].format(sub_name))
 
         try:
+            rules = [rule async for rule in sub.rules]
+
+            if len(rules) == 0:
+                await msg.edit(content=i18n["rules"]["no_rules"].format(sub_name))
+                return
+
             async for rule in sub.rules:

```

Comment:
Not too sure of how `async` works, but wouldn't you be able to use the newly defined `rules` variable here to avoid the retrieval of the rules in an asynchronous manner twice?

Reply:
Absolutely. I extracted the rules later on and forgot to change it there, highlighting that `rules` and `pi_rules` share too much code. I fixed that now.